From dad44178cec1ed329f3fa499ccdbb310869a91c6 Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Wed, 23 Oct 2019 13:40:08 +0300
Subject: [PATCH 38/39] BACKMERGE to 0016, partially restore.

 Exposing "struct malloc_type" internals is useful even under kernel
 structures namespace. In vmstat(8) where this will allow to drop
 dependency on the <sys/malloc.h> header.
---
 sys/sys/objcache.h | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/sys/sys/objcache.h b/sys/sys/objcache.h
index 53a31dee2f..483edc0c7a 100644
--- a/sys/sys/objcache.h
+++ b/sys/sys/objcache.h
@@ -33,13 +33,15 @@
 #ifndef _SYS_OBJCACHE_H_
 #define _SYS_OBJCACHE_H_
 
-#if defined(_KERNEL) || defined(_KERNEL_STRUCTURES)
-
 #ifndef _SYS_TYPES_H_
 #include <sys/types.h>
 #endif
 
-struct malloc_type;
+#if defined(_KERNEL) || defined(_KERNEL_STRUCTURES)
+
+#ifndef _SYS__MALLOC_H_
+#include <sys/_malloc.h>
+#endif
 
 #define OC_MFLAGS	0x0000ffff	/* same as malloc flags */
 
@@ -63,9 +65,9 @@ struct objcache
 			 objcache_alloc_fn *alloc, objcache_free_fn *free,
 			 void *allocator_args);
 struct objcache
-	*objcache_create_simple(struct malloc_type *mtype, size_t objsize);
+	*objcache_create_simple(malloc_type_t mtype, size_t objsize);
 struct objcache
-	*objcache_create_mbacked(struct malloc_type *mtype, size_t objsize,
+	*objcache_create_mbacked(malloc_type_t mtype, size_t objsize,
 			  int cluster_limit, int nom_cache,
 			  objcache_ctor_fn *ctor, objcache_dtor_fn *dtor,
 			  void *privdata);
@@ -85,7 +87,7 @@ void	 objcache_destroy(struct objcache *oc);
  */
 struct objcache_malloc_args {
 	size_t		objsize;
-	struct malloc_type *mtype;
+	malloc_type_t	mtype;
 };
 
 #ifdef	_KERNEL
-- 
2.23.0

